nemo/collections/asr/modules/sortformer_modules.py

```diff
@@ -12,68 +12,145 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import math
+from dataclasses import dataclass
+from typing import List, Tuple
+
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
 
 from nemo.core.classes.exportable import Exportable
 from nemo.core.classes.module import NeuralModule
+from nemo.utils import logging
 
 __all__ = ['SortformerModules']
 
 
+def concat_and_pad(embs: List[torch.Tensor], lengths: List[torch.Tensor]):
```

Comment:
There is another function with similar functionality and same name in models/sortformer_diar_models.py, is it possible to combine them and make this function reusable for other tasks?

Reply:
This is removed.